README.md

```diff
@@ -34,9 +34,11 @@ jobs:
       - uses: julia-actions/setup-julia@latest
       - name: VersionVigilante.main
         id: versionvigilante_main
+        env:
+         github-token: ${{secrets.GITHUB_TOKEN}}
         run: |
           julia -e 'using Pkg; Pkg.add("VersionVigilante")'
-          julia -e 'using VersionVigilante; VersionVigilante.main("https://github.com/${{ github.repository }}")'
+          julia -e 'using VersionVigilante; VersionVigilante.main("https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}")'
```

Comment:
I don't think this should be the default. It suggests that VersionVigilante requires access to the token which is not the case (or is it?). There are people who are concerned about giving the token to untrusted code and it's possible they'd see it and then not use the workflow due to that.

Reply:
That's a very good point.

Is there a way for us to automatically figure out if a repository is private or public?

Reply:
Not sure if there's a nice way, but that also wouldn't really make a difference because it'd still have the token in the workflow. I think having two examples or that line duplicated and commented out would work well.

Alternatively, but I don't know if that would require a major but nonbreaking rewrite, would it be possible to use the locally cloned repo? That would avoid this issue entirely.

Reply:
> Alternatively, but I don't know if that would require a major but nonbreaking rewrite, would it be possible to use the locally cloned repo? That would avoid this issue entirely.

This isn't really possible. You need to get the latest master branch and get its version number. And you can't just do `git checkout master` in the locally cloned repo - if the PR was made from a fork, and you do `git checkout master`, then you'll get the master branch of the forked repo. But you don't want to the master branch of the forked repo, you want the master branch of the upstream repo. It is very common for people to have forks in which the master branch in the fork is many commits behind the master branch of upstream.

> Not sure if there's a nice way, but that also wouldn't really make a difference because it'd still have the token in the workflow. I think having two examples or that line duplicated and commented out would work well.

Yeah, probably it is best to just add the commented out lines, instead of making this the default.

Reply:
I see, makes sense. Thanks for explaining :)

For private repos it might still be a useful tokenless alternative if there's demand for it. I'd be surprised if there are many PRs from forks in private repos. But that's probably out of scope for this PR